fm.c

```diff
@@ -53,12 +53,10 @@ void FM_Prescaler(fm_prescaler_t *chip)
 {
     if (!chip->input.phi)
     {
-        int ic_check;
-
         chip->ic_latch[0] = chip->ic_latch[1] << 1;
         chip->ic_latch[0] |= chip->input.ic;
-
-        ic_check = (chip->ic_latch[1] & 0x800) == 0 && chip->input.ic;
+        
+        const int ic_check = ((!((chip->ic_latch[1] & 0x800) & 1)) ^ chip->input.ic > 0);

```

Comment:
this doesn't look right

Reply:
this is using bitwise tricks to get rid of the compare; & 1 will always be either 0/1, the xor replaces the logical and. this translates to faster instructions over two compares. the false statement may actually be over kill since i check if its greater than zero.

The idea i'm going for is faster instructions; AND and XOR are faster instructions.

Reply:
`((chip->ic_latch[1] & 0x800) & 1)` part will be always be zero, and I'm not sure how xor can replace and